<documentation title="Escaped Not Translated Text">
    <standard>
    <![CDATA[
    Text intended for translation needs to be wrapped in a localization function call.
    This sniff will help you find instances where text is escaped for output, but no localization function is called, even though an (unexpected) text domain argument is passed to the escape function.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: esc_html__() used to translate and escape.">
        <![CDATA[
echo <em>esc_html__</em>( 'text', 'domain' );
        ]]>
        </code>
        <code title="Invalid: esc_html() used to only escape a string intended to be translated as well.">
        <![CDATA[
echo <em>esc_html</em>( 'text', 'domain' );
        ]]>
        </code>
    </code_comparison>
</documentation>
