<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Global Variables Override"
    >
    <standard>
    <![CDATA[
    WordPress global variables should not be overridden.

    Two WordPress global variables, `$content_width` and `$wp_cockneyreplace`, are 
    specifically intended to be overridden by themes/plugins and are exempt from this rule.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Different variable name.">
        <![CDATA[
<em>$my_query</em> = new WP_Query( $args );


<em>$GLOBALS['my_data']</em> = "some data";
        ]]>
        </code>
        <code title="Invalid: Global variable overridden.">
        <![CDATA[
<em>global $wp_query;</em>
<em>$wp_query</em> = new WP_Query( $args );

<em>$GLOBALS['wp_query']</em> = new WP_Query( $args );
        ]]>
        </code>
    </code_comparison>
</documentation>
